src/stats.rs.in

```diff
@@ -0,0 +1,48 @@
+#[derive(Debug, PartialEq, Eq, Deserialize, Clone)]
+pub struct Period {
+    pub day: String,
+    pub views: u32,
+    pub visitors: u32,
+    pub reach: u32,
+    pub reach_subscribers: u32,
+    pub subscribed: u32,
+    pub unsubscribed: u32,
+    pub sex: Vec<Demography>,
+    pub age: Vec<Demography>,
+    pub sex_age: Vec<Demography>,
+    pub cities: Vec<City>,
+    pub countries: Vec<Country>,
+}
+
+#[derive(Debug, PartialEq, Eq, Deserialize, Clone)]
+pub struct Demography {
+    pub visitors: u32,
+    pub value: String,
```

Comment:
The `value` field actually have limited number of variants, depending on field it is used in. E.g. for `sex` field it can have the following type:

```rust
enum_str! { DemoSex {
    Male = "m",
    Female = "f",
}}
```

For the `age` field the type can be:

```rust
enum_str! { DemoAgeRange {
    _12_18 = "12-18",
    _18_21 = "18-21",
    _21_24 = "21-24",
    _24_27 = "24-27",
    _27_30 = "27-30",
    _30_35 = "30-35",
    _35_45 = "35-45",
    _45_100 = "45-100".
}
```

And for `city`:

```rust
pub enum DemoCity {
    CityId(Id),
    Other,
}
```

And then the `Demography` type would be:
```rust
pub struct Demography<T: Deserialize + Copy + Debug + Eq> {
    pub visitors: u32,
    pub: value<T>,
}
```